### ISSUE TITLE
Staging site is being used as production/live site

### ISSUE THREAD

**Issue by LucyGwilliamAdmin** [sustainabledevelopment-rwanda/sdg-indicators#42]
Staging site link has been promoted instead of live site link. What is the best way to fix this?

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning any suggestions on best way to fix this? 
Just a few ideas so far:
* swap staging site and live site (I think this will be quite complicated to do and I think it's quite strange to have the URL to main page of site as [MAIN_URL]/[PAGE] but will means those who link has been shared to will still work and link in any publications will still work)
* Change staging site to new URL and create new page on Rwanda NRP called SDG Indicators which lives at https://sustainabledevelopment-rwanda.github.io/sdg-indicators (slightly easier fix than first choice and main link will still work but links to other pages (e.g. https://sustainabledevelopment-rwanda.github.io/sdg-indicators/1-1-1) won't work

**Comment by LucyGwilliamAdmin** [comment]
@AnnCorp @Norric1Admin (can't tag Savvas or Mike as don't think they have even write access)
Just making you aware of this issue

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin That's a tricky one. To answer your question about Google Analytics - yes, since the tracking code is controlled in the config file, you can have different codes in each.

I think I prefer your second option though. The new page could be a redirect to the homepage?

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning thanks so much for your input

With the second approach could all pages including 'sdg-indicators' in URL redirect to homepage e.g. indicator pages, etc.

And how would this work technically?

**Comment by brockfanning** [comment]
You could use the "redirect-from" plugin. See [here](https://github.com/jekyll/jekyll-redirect-from#usage). It would be a long list, but it should work in theory.  After adding the redirect-from plugin you would edit the [homepage](https://github.com/sustainabledevelopment-rwanda/sdg-indicators/blob/develop/_pages/index.md) and add something like:
```
redirect_from:
  - /sdg-indicators/
  - /sdg-indicators/1/
  - /sdg-indicators/2/
  etc...
```

**Comment by Norric1Admin** [comment]
A redirect sounds like a good approach, as we avoid having to switch the repos somehow, and the promotions using the wrong link won't have to be recalled.

To consider: if /sdg-indicators redirects to the live site in future, this makes the staging site unusable. NISR will have to set up feature branches so that they have somewhere to trial updates.

Savvas is investigating this for the UK, and we'll forward his notes once available.

**Comment by LucyGwilliamAdmin** [comment]
Yes, in the suggestion I mentioned that the staging site would have to be moved to a new address so don't necessarily need feature branches as can test changes on staging site. Don't think moving staging site to new address should be too difficult though

**Comment by mikebolton10** [comment]
Has anything come of this? I have found out the the link to the staging site was shared only shared on the NISR website, and not widely outside of that.  So it might be that people aren't using that link and we can push to live site, change the URL being advertised to the livesite and then move back to the standard staging/live set up. 

I think that's the best/easiest option but it's dependent on whether people are using the link to the staging site that's been published.  So Lucy's google analytics will make interesting reading.  What's the timeframe on those @LucyGwilliamAdmin ?

**Comment by LucyGwilliamAdmin** [comment]
Hi @mikebolton10 , that's good news! That would definitely be the best way forward if possible.

I set up Google Analytics on Friday last week (I think) and since then the staging site has had 21 users visiting the site (with 15 of those being yesterday - maybe lots of people were accessing as you were there?

Since traffic is usually lower on weekends, you might want to wait a bit longer to say for definite, but it seems like the staging site hasn't had many visits at all. I will finish setting up the analytics for the live site now.

**Comment by Norric1Admin** [comment]
A bit of input on this - this repos were set up incorrectly from the start. https://github.com/sustainabledevelopment-uk should not contain 'sdg-indicators', and I believe this is a major cause of the problem (see https://github.com/sustainabledevelopment-uk for comparison).

**Comment by Norric1Admin** [comment]
In my experience this has been working fine, just with the URLs in reverse. I agree that we should hold off any transfers until @mikebolton10 returns from overseas. I recommend setting up an equivalent of https://github.com/sdg-kyrgyzstan/ for Rwanda.

**Comment by mikebolton10** [comment]
Yes currently we can add files to develop branch, which show on the staging site, and then push to master branch which then show on the live site.  So in that respect it's working

**Comment by Norric1Admin** [comment]
Actually an ideal situation would be to use Savvas' guidance to help them set up feature branches (requires them to have a server, but NISR should be able to provide something), and then we can redirect the Staging site (.../sdg-indicators) to the Live site. Feature branches will then replace Staging.

This guidance isn't written yet, but it should be coming sometime this month.

**Comment by SavvasStephanides** [comment]
@Norric1Admin Thanks Chris. I've been playing around trying to implement a solution for hosting feature branches and I have a work in progress https://github.com/savvas-sdg/feature-branch-deployment

**Comment by mikebolton10** [comment]
FYI - the link to the NRP that appears on the NISR website has now been changed so it links to the live site, as opposed to the staging site as it was doing before.  

It still doesn't solve the problem, as anyone that may have bookmarked the page previously will still go to the staging site.  But at least any new users will not now make that same mistake.

**Comment by LucyGwilliamAdmin** [comment]
@mikebolton10 the more I think about it, the more I think we should move the staging site to a diff address (sdg-rwanda.github.io/sdg-indicators) - I think it will make it clearer which one is the staging site and avoid confusion in the future. 

I've looked into it a bit more and think moving the staging site and repos will be pretty easy. 

Do you want to check with Rwanda team while you're out there if this is what they want? 

Just as will mean they will be uploading files to diff place (sdg-rwanda/sdg-data instead of sustainabledevelopment-rwanda)

**Comment by AnnCorp** [comment]
@mikebolton10 @LucyGwilliamAdmin is this still an issue or could this be closed?
